src/include/migraphx/algorithm.hpp

```diff
@@ -76,6 +96,53 @@ void group_unique(Iterator start, Iterator last, Output out, Predicate pred)
     }
 }
 
+template <class Iterator, class Predicate, class Output>
+void group_find(Iterator start, Iterator last, Predicate pred, Output out)
+{
+    start = std::find_if(start, last, pred);
+    while(start != last)
+    {
+        auto it = std::find_if_not(start, last, pred);
+        out(start, it);
+        start = std::find_if(it, last, pred);
+    }
+}
```

Comment:
To me it seems like this is better implemented with just single `for` or `while` loop. 
e.g. 
```
auto pred_it = last;
auto neg_pred_it = last;
curr_it = start;
while(curr_it != last) {
    if(pred(curr_it) {
        pred_it = curr_it;
    } else if(pred_it != last) { // pred_it must be set 
        neg_pred_it = curr_it;
    }
    if(pred_it != last && neg_pred_it != last) {
        out_fun(pred_it, neg_pred_it)
        pred_it = last;
        neg_pred_it = last;
    }
    curr_it++;
}
```

Reply:
Using `find_if` is easier to read though and it matches the pattern of how the other group algorithms work.